src/Controls/src/Core/Platform/AlertManager/AlertManager.Windows.cs

```diff
@@ -244,16 +245,8 @@ static async Task<bool> ShowAlert(ContentDialog alert)
 				return null;
 			}
 
-			bool PageIsInThisWindow(Page page)
-			{
-				var window = page?.Window;
-				var platformWindow = window?.MauiContext.GetPlatformWindow();
-
-				if (window is null || platformWindow is null)
-					return false;
-
-				return platformWindow == Window;
-			}
+			bool PageIsInThisWindow(Page page) =>
+				page?.Window == VirtualView;
```

Comment:
The alert manager already knows about the window, so this is now possible. The iOS code had everything set up, but Windows just needed an extra constructor argument for the helper to simplify all this.